### ISSUE TITLE
Improvement suggestions for protocol paper

### ISSUE THREAD

**Issue by syncu2** [ReSpeak/tsdeclarations#9]
I finally managed to get my Python client to work with the new protocol (many thanks for your awesome work in researching and documenting it) and would like to make a few suggestions for improvements on the protocol paper to make things more clear for people who want to implement the protocol, too.

- In section 3.2.2.4, it says "Teamspeak uses Curve25519 for all cryptographic operations.". This might be misleading, because TS is using AES for encryption and Prime256 + Ed25519 curves for signing/authenticating.
- On multiple places, Curve25519 is mentioned with regards to key pairs and points, but I think we are actually using Ed25519, right?
- It should be mentioned that the `clientek` packet already counts as Packet ID 1 rather than 0.
- For deriving the key from the licenses, it should be explained that the Ed25519 private key is decoded like this:
```
    buffer := sha512(licenseBlock);
    // clamping
    buffer[0] &= ~0x07;
    buffer[31] &= 0x7F;
    buffer[31] |= 0x40;
    privateKey := buffer interpreted as a little-endian big int
```
- It should be mentioned that the ECDSA signatures (`proof` parameter) must be calculated/verified using SHA256 and encoded in DER format (both are no default, at least in the library I am using).
- Small typo in section 3.2.2.4: Should be `SharedIV[10-64]` instead of `SharedIV[10-56]`

General suggestions:
- Explain how the values in the `setconnectioninfo` reply are calculated. I tried implementing the reply, but the server does not seem to accept my values (connection info window keeps "Loading").

**Comment by Splamy** [comment]
Thank you for your time and feedback. I will try to include as much as I can.
- Yes, this clarifications seems reasonable
- We've had that discussion too :) and we came to the conclusion, that Curve25519 is the correct term. Ed25519 describes the the EDDSA signing operations *on* the Curve25519. (Similar what ECDSA is used on for e.g. P-256). [See here for more explanation.](https://security.stackexchange.com/questions/50878/ecdsa-vs-ecdh-vs-ed25519-vs-curve25519). If you think we misinterpreted this or this is indeed incorrect feel free to write back.
- Correct.
- I was still curious about this since this operations 'claps' the compressed number to `2^252+...` somehow our rust implementation didn't need that, so I didn't know if it was a +/- sign error or actually needed. I will add a note though.
- Yup, this is also correct.
- Nice catch

- Usually we try to refer to the Messages.txt for command setups and datatypes. I will post an example here and when we find some problems i will also include those in the proto paper.

Example:
```
setconnectioninfo connection_ping=29.0000 connection_ping_deviation=9.1754 connection_packets_sent_speech=0 connection_packets_sent_keepalive=17 connection_packets_sent_control=39 connection_bytes_sent_speech=0 connection_bytes_sent_keepalive=715 connection_bytes_sent_control=2987 connection_packets_received_speech=0 connection_packets_received_keepalive=17 connection_packets_received_control=35 connection_bytes_received_speech=0 connection_bytes_received_keepalive=697 connection_bytes_received_control=9345 connection_server2client_packetloss_speech=0.0000 connection_server2client_packetloss_keepalive=0.0000 connection_server2client_packetloss_control=0.0000 connection_server2client_packetloss_total=0.0000 connection_bandwidth_sent_last_second_speech=0 connection_bandwidth_sent_last_second_keepalive=83 connection_bandwidth_sent_last_second_control=0 connection_bandwidth_sent_last_minute_speech=0 connection_bandwidth_sent_last_minute_keepalive=77 connection_bandwidth_sent_last_minute_control=363 connection_bandwidth_received_last_second_speech=0 connection_bandwidth_received_last_second_keepalive=81 connection_bandwidth_received_last_second_control=0 connection_bandwidth_received_last_minute_speech=0 connection_bandwidth_received_last_minute_keepalive=75 connection_bandwidth_received_last_minute_control=1144
```

**Comment by syncu2** [comment]
Yeah I am not really an expert on elliptic curves, but reading on the web about it, it seemed to me that Ed25519 was a curve that can be derived from Curve25519 but not the same one. I was not sure either, though. The libraries are not documented very well, even less the low-level stuff for point algebra...
